src/qiboml/models/decoding.py

```diff
@@ -283,3 +286,38 @@ def output_shape(self) -> tuple[int, int]:
     @property
     def analytic(self) -> bool:  # pragma: no cover
         return False
+    
+@dataclass
+class VQLS(QuantumDecoding):
+    """Decoder for the Variational Quantum Linear Solver (VQLS).
```

Comment:
You could add a reference to Carlos' paper here. Or explaining a bit more about the utility of this decoder :) 
This could also help the users to understand the notation of the arguments (why you have a target state, why there is a matrix `A`, etc).